http.go

```diff
@@ -112,11 +129,65 @@ func (w *logResponseWriter) WriteString(data string) (int, error) {
 	return n, err
 }
 
+func (w *logResponseWriter) Status() int {
+	return w.status
+}
+
+func (w *logResponseWriter) Size() int64 {
+	return w.size
+}
+
+type logResponseWriter2 struct {
+	StdResponseWriter2
+	status int
+	size   int64
+}
+
+func (w *logResponseWriter2) WriteHeader(status int) {
+	w.status = status
+	w.StdResponseWriter2.WriteHeader(status)
+}
+
+func (w *logResponseWriter2) Write(data []byte) (int, error) {
+	n, err := w.StdResponseWriter2.Write(data)
+	w.size += int64(n)
+	return n, err
+}
+
+func (w *logResponseWriter2) WriteString(data string) (int, error) {
+	n, err := w.StdResponseWriter2.WriteString(data)
+	w.size += int64(n)
+	return n, err
+}
+
+func (w *logResponseWriter2) Status() int {
+	return w.status
+}
+
+func (w *logResponseWriter2) Size() int64 {
+	return w.size
+}
+
+func createLogWriter(w http.ResponseWriter) (http.ResponseWriter, logWriter) {
+	if srw1, ok := w.(StdResponseWriter); ok {
+		t := &logResponseWriter{srw1, http.StatusOK, 0}
+		return t, t
+	}
+
+	if srw2, ok := w.(StdResponseWriter2); ok {
+		t := &logResponseWriter2{srw2, http.StatusOK, 0}
+		return t, t
+	}
+
+	panic("unexpected ResponseWriter implementation")
+}
+
 // ServeHTTP implements http.Handler interface.
 func (s *HTTPServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	startTime := time.Now()
 
-	lw := &logResponseWriter{w.(StdResponseWriter), http.StatusOK, 0}
+	w, lw := createLogWriter(w)
```

Comment:
I'm not entirely sure of the role of `lw` here. If we reach this point, `w` is guaranteed to be of `logResponseWriter` or `logResponseWriter2` type so we can safely assert it has the `status` and `size` fields.

Reply:
Yes they have.  But without `lw`, how do you access those fields?

Reply:
Oh I see, `w` in this context was implicitly cast to a `ResponseWriter`. Got it.